.github/workflows/tests.yml

```diff
@@ -0,0 +1,99 @@
+name: Run unit and integration tests
+
+on:
+  push:
+    branches:
+      - master
+    paths-ignore:
+      - 'docs/**'
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+
+
+jobs:
+  test-conda-build:
+    runs-on: ${{ matrix.os }}
+    defaults:
+      run:
+        shell: bash -l {0}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]
+        python-version: ['3.7', '3.8']
+        conda-version: [release]
+        test-type: [serial, parallel]
+        include:
+          - os: ubuntu-latest
+            python-version: '3.9'
+            conda-version: canary
+            test-type: serial
+          - os: ubuntu-latest
+            python-version: '3.9'
+            conda-version: canary
+            test-type: parallel
+    env:
+      pytest-replay: --replay-record-dir=pytest-replay/ --replay-base-name=Linux-${{ matrix.conda-version }}-Py${{ matrix.python-version }}
+
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Setup Miniconda
+        uses: conda-incubator/setup-miniconda@v2
+        with:
+          condarc-file:  ci/github/.condarc
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install Conda-Canary if canary version
+        if: matrix.conda-version == 'canary'
+        run: |
+          conda install -y -c conda-canary conda
+
+      - name: Install default Conda if release version
+        if: matrix.conda-version == 'release'
+        run: |
+          conda install -y conda
+
+      - name: Source Scripts
+        run: |
+          source ci/azurepipelines/install_conda_build_test_deps
+          pip install -e .
+          conda info -a
+          source ci/github/github_setup_pytest_replay
+        env:
+          PYTHON_VERSION: ${{ matrix.python-version }}
+          CONDA_VERSION: ${{ matrix.conda-version }}
+
+      - name: Run Serial tests
+        if: matrix.test-type == 'serial'
+        run: |
+          pytest --color=yes -vv -n 0 --basetemp /tmp/cb_serial --cov conda_build --cov-report xml -m "serial" tests --test-run-title="${{ matrix.os }}-SerialTests" --test-run-title="${{ matrix.os }}-SerialTests-${{ matrix.conda-version }}-Py${{ matrix.python-version }}" $PYTEST_REPLAY
+        env:
+          PYTHON_VERSION: ${{ matrix.python-version }}
+          CONDA_VERSION: ${{ matrix.conda-version }}
```

Comment:
I don't think these are used?

Reply:
They are used as part of the "--test-run-title"

Reply:
But you didn't use the env vars above in `--test-run-title`?

Reply:
Oh yeah... huh? How'd that happen...  yeah what the jank. Let me fix that. I am so confused because this goes against what I said before about how I have to call it in the "env:" to use it in the script. Let me research this if I am remembering it right. I guess it has been 2 weeks since I looked at this.

Reply:
Welp it appears I got my head all muddled. I removed all the "env:" that duplicate a variable that has already been declared. This is exactly why we do reviews lol